### PR TITLE
Update is_root and only_on_root functions.

### PR DESCRIPTION
**Pull request by brittonsmith**
This updates the `is_root` and `only_on_root` functions for the new configuration.

## PR Checklist

- [ ] New features are documented, with docstrings and narrative docs
- [ ] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

**Review by matthewturk — APPROVED** [review]

**Review by munkm — APPROVED** [review]
Amazing branch name.

**Review by neutrinoceros — APPROVED** [review]
+1 @munkm !

File: yt/funcs.py [hunk 1]
```diff
@@ -429,10 +429,10 @@ def is_root():
     """
     from yt.config import ytcfg
 
-    cfg_option = "__topcomm_parallel_rank"
+    cfg_option = "topcomm_parallel_rank"
```
Comment (cphyc) [inline yt/funcs.py:432]:
```suggestion
```

File: yt/funcs.py [hunk 2]
```diff
@@ -429,10 +429,10 @@ def is_root():
...
     if not ytcfg.get("yt", "internals", "parallel"):
         return True
-    if ytcfg.get("yt", cfg_option) > 0:
+    if ytcfg.get("yt", "internals", cfg_option) > 0:
         return False
     return True
```
Comment (cphyc) [inline yt/funcs.py:437]: Couldn't be simplified as (assuming above-mentioned change is accepted)
```suggestion
    return ytcfg.get("yt", "internals", "topcomm_parallel_rank") == 0
```
unless `ytcfg.get("yt", "internals", "topcomm_parallel_rank")` can take negative values or non-integers?

**Review by cphyc — APPROVED** [review]
I re-read all the calls to `ytcfg` and convinced myself these two functions are the only one that needed an update.

I left two minor comments, but I'd fine to merge this as is.

File: yt/funcs.py (hunk 2 above)
Comment (brittonsmith) [reply yt/funcs.py:437]: I don't think so. Good suggestion.

**Pull request merged**